code/modules/reagents/machinery/thermometer.dm

```diff
@@ -0,0 +1,130 @@
+	splash_sub(reagents, C, -1, user)
+	user.drop_item(src)
+	playsound(src, "shatter", 70, 1)
+	new /obj/item/weapon/broken_thermometer(get_turf(src))
+	qdel(src)
+
+/obj/item/weapon/thermometer/preattack(obj/target, mob/user, proximity_flag, click_parameters)
+	if(proximity_flag == 0) // not adjacent
+		return
+
+	if(istype(target, /obj/machinery/bunsen_burner))
+		var/obj/machinery/bunsen_burner/B = target
+		if(B.held_container)
+			target = B.held_container
+
+	if(!target.reagents)
+		return
+
+	if(ismob(target))
+		var/mob/living/carbon/human/C = target
+		if(user.a_intent != I_HELP) //Fuck!
+			crit_fail(C, user)
+		else
+			if(user.zone_sel.selecting == "mouth" && !C.check_body_part_coverage(MOUTH))
+				visible_message("<span class = 'notice'>\The [user] starts taking \the [C]'s temperature using \the [src].</span>",\
+				"<span class = 'notice'>You start taking the temperature of \the [user].</span>")
+				if(do_after(user, src, 20))
+					last_temperature_string = measure_human_temperature(C)
+					update_icon()
+					to_chat(user, "<span class = 'notice'>The temperature reads: [last_temperature_string]</span>")
+		return 1
+
+	to_chat(user, "<span class = 'notice'>You measure the temperature of \the [target] with \the [src].</span>")
+	last_temperature_string = measure_obj_temperature(target)
+	update_icon()
+	to_chat(user, "<span class = 'notice'>The temperature reads: [last_temperature_string]</span>")
+
+	return 1
+
+/obj/item/weapon/thermometer/attack_self(mob/user)
+	to_chat(user, "<span class = 'notice'>Last recorded temperature: [last_temperature_string]</span>")
+
+/obj/item/weapon/thermometer/proc/measure_human_temperature(mob/living/carbon/human/C)
+	last_temperature = C.bodytemperature
+	return "[round(last_temperature-273.15,5)] C"
+
+/obj/item/weapon/thermometer/proc/measure_obj_temperature(obj/target)
+	last_temperature = target.reagents.chem_temp
+	return "[round(last_temperature-273.15, 5)] C"
+
+/obj/item/weapon/thermometer/electronic
+	name = "electronic thermometer"
+	desc = "An electronic thermal probe, boasting greater precision and less mercury than its analogue counterparts."
+	icon_state = "therm_digi_1"
+	origin_tech = Tc_ENGINEERING + "=3;" + Tc_BIOTECH + "=3"
+
+	starting_materials = list(MAT_IRON = 1000, MAT_GLASS = 400)
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_GLASS
+
+/obj/item/weapon/thermometer/electronic/initial_thermometer()
+	return
+
+/obj/item/weapon/thermometer/electronic/update_icon()
+	return
+
+/obj/item/weapon/thermometer/electronic/crit_fail()
+	return
+
+/obj/item/weapon/thermometer/electronic/measure_human_temperature(mob/living/carbon/human/C)
+	last_temperature = C.bodytemperature
+	return "[last_temperature-273.15] C"
+
+/obj/item/weapon/thermometer/electronic/measure_obj_temperature(obj/target)
+	last_temperature = target.reagents.chem_temp
+	return "[last_temperature-273.15] C"
+
+
+/obj/item/weapon/broken_thermometer
```

Comment:
Is there any reason for a broken thermometer to be its own item, rather than just setting all the appropriate vars on a regular thermometer to make it look and behave this way?

Reply:
No unforseen behaviour from leftover variables and redefined procs. Functionally the same as a broken bottle, which is /obj/item/weapon/broken_bottle rather than /obj/item/weapon/reagent_containers/food/drinks/bottle/broken

Reply:
Well, my suggestion was more along the lines of it being an `/obj/item/weapon/reagent_containers/food/drinks/bottle` that had its variables changed to simply look and act like a broken bottle.
Still, fair enough.